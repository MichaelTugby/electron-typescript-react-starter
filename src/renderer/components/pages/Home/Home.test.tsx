// import React from "react";
// import { cleanup, render, fireEvent } from "@testing-library/react";
// import Home from "./Home";

// import { Provider as StoreProvider } from "react-redux";
// import { ThemeProvider } from "@emotion/react";

// import store from "~/renderer/store";
// import theme from "~/renderer/theme";

// describe("Home", () => {
//   afterEach(cleanup);

//   it("snapshot has not changed", () => {
//     const { container } = render(
//       <StoreProvider store={store}>
//         <ThemeProvider theme={theme}>
//           <Home />
//         </ThemeProvider>
//       </StoreProvider>
//     );
//     expect(container).toMatchSnapshot();
//   });

//   it("fires SUBTRACT_COUNTER dispatch method on decrease", () => {
//     const spy = jest.spyOn(store, "dispatch");
//     const { getByText } = render(
//       <StoreProvider store={store}>
//         <ThemeProvider theme={theme}>
//           <Home />
//         </ThemeProvider>
//       </StoreProvider>
//     );
//     const subtract = getByText("-");
//     fireEvent.click(subtract);
//     expect(spy).toHaveBeenCalledWith({ type: "SUBTRACT_COUNTER" });
//   });

//   it("fires ADD_COUNTER dispatch method on increase", () => {
//     const spy = jest.spyOn(store, "dispatch");
//     const { getByText } = render(
//       <StoreProvider store={store}>
//         <ThemeProvider theme={theme}>
//           <Home />
//         </ThemeProvider>
//       </StoreProvider>
//     );
//     const subtract = getByText("+");
//     fireEvent.click(subtract);
//     expect(spy).toHaveBeenCalledWith({ type: "ADD_COUNTER" });
//   });

//   it("fires toggleDarkMode method on toggle dark mode button click", () => {
//     const spy = jest.fn();
//     global.api = {
//       ...global.api,
//       send: spy,
//     };
//     const { getByText } = render(
//       <StoreProvider store={store}>
//         <ThemeProvider theme={theme}>
//           <Home />
//         </ThemeProvider>
//       </StoreProvider>
//     );
//     const toggleDarkModeBtn = getByText("Toggle Dark Mode");
//     fireEvent.click(toggleDarkModeBtn);
//     expect(spy).toHaveBeenCalledWith("dark-mode:toggle");
//   });
// });
