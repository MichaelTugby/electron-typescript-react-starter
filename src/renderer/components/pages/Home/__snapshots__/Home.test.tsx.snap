// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot has not changed 1`] = `
<div>
  <div
    id="home"
  >
    <h3
      class="css-v6lo72-Home"
    >
      Simple Counter
    </h3>
    <div
      class="counter"
      id="home__counter"
    >
      <button
        class="counter__decrease-btn css-1msykvu-Button"
      >
        -
      </button>
      <span
        class="counter__text css-10wtsar-Counter"
      >
        1
      </span>
      <button
        class="counter__increase-btn css-1msykvu-Button"
      >
        +
      </button>
    </div>
    <button
      class="button css-50ddk5-Button"
    >
      Toggle Dark Mode
    </button>
  </div>
</div>
`;
